crates/shim-sgx/src/handler/mod.rs

```diff
@@ -996,4 +1144,117 @@ impl<'a> Handler<'a> {
 
         Ok(())
     }
+
+    fn handle_page_fault(&mut self, addr: usize, error_code: PageFaultErrorCode) -> Result<(), ()> {
+        let tid = self.tcb.tid;
+        let addr = addr & !(Page::SIZE - 1);
+
+        debugln!(self, "[{tid}] handle_page_fault: {addr:#x} {error_code:?}");
+
+        // check for non-valid flags
```

Comment:
Would it be more future-proof to check for valid flags instead?

Reply:
good point! Thanks!